### PR TITLE
Yao - Add less-than-minimum hours logged colors on Weekly Summaries Reports page

### PR DESCRIPTION
**Pull request by yaow62023**
# Description
Add less-than-minimum hours logged colors on Weekly Summaries Reports page (WIP Yao)
Other Links → Reports → Weekly Summaries Reports
If someone has done less than 49% of their hours, the whole Hours Logged row and hours should be red
If someone has done between 50-99% of their hours, the whole Hours Logged row and hours should be bright green
This is needed so it is clear to the Admins who these people are. Red people don’t get credited for their work for the week. Green and black people do. 

## Main changes explained:
- Add different text color for different hours logged.
- For less then 49% hours logged display as red, for 50 - 99% display as green, for 100% display as black. 

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to dashboard→ Report→ Weekly summary report
6. use a Owner account to add different Tangible Time Entry
7. test the color display correctly 

### less then 49% hours logged:
Screenshot 2024-02-24 at 11 56 22 AM

### 50 - 99% hours logged:
Screenshot 2024-02-24 at 11 57 26 AM

### 100% hours logged:
Screenshot 2024-02-24 at 11 55 58 AM

### CODE REVIEW

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I reviewed your PR and hours are being color-coded correctly. Please refer the screenshots attached.

PR 1990 black
PR 1990 green
PR 1990 red

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 1]
```diff
@@ -172,6 +172,18 @@ function EmailsList({ summaries, auth }) {
...
+
+  if (percentage < 49) {
+    return 'red';
+  } else if (percentage >= 50 && percentage < 100) {
+    return 'lightgreen'; // Bright green color
+  } else {
+    return 'black'; // Default color
+  }
+}
+
```
Comment (FyisFe) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:186]: 49 <= `percentage` < 50 case is not covered, therefore it becomes black.

**Review by FyisFe — CHANGES_REQUESTED** [review]
I tested it and found the following bug. When the percentage is 49%, it displays black

image

**Review by XiaohanMeng — CHANGES_REQUESTED** [review]
Hi, I tested this PR and found the same issue as FyisFe: at 49% hours logged, the color turns black. Please refer to the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/aed25133-9a9e-45f9-b27f-df08a89901e9

File: src/components/WeeklySummariesReport/FormattedReport.jsx (hunk 1 above)
Comment (yaow62023) [reply src/components/WeeklySummariesReport/FormattedReport.jsx]: Hi, thanks for discover this bug. I have change the logic of this function. You can test it again. Thank you so much!

**Comment by yaow62023** [comment]
> Hi, I tested this PR and found the same issue as FyisFe: at 49% hours logged, the color turns black. Please refer to the attached video.
> 
>  1990-result.mov

**Comment by dhairya-mehta** [comment]
Hey , I tested and verified the following PR
PR 1990

**Review by ClemarNunes — DISMISSED** [review]
I tested your PR and worked as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/853c3061-84c8-4b2d-97c4-9f934399053f

**Comment by one-community** [comment]
Please make the green darker color code 0B6623 so it stands out more

**Review by SRamen1999 — COMMENTED** [review]
After testing this PR all the color changes for hours logged work as intended. However I also encountered the same bug where 4.90 / 10 is not changed properly

bug
Screen Shot 2024-04-19 at 10 11 48 AM


 less then 49% hours logged 
Screen Shot 2024-04-19 at 10 15 30 AM



50 - 99% hours logged
Screen Shot 2024-04-19 at 10 16 19 AM


100% hours logged work as intended. 
Screen Shot 2024-04-19 at 10 15 00 AM

**Review by metaphor987 — DISMISSED** [review]
Hi Yao, I've tested with different logged times, and the functionalities work as expected. Nice work! Below are the screenshots showing the colors for some edge cases of logged time.
Screenshot 2024-04-21 at 1 26 23 AM
Screenshot 2024-04-21 at 1 31 49 AM
Screenshot 2024-04-21 at 1 30 26 AM
Screenshot 2024-04-21 at 1 40 13 AM
Screenshot 2024-04-21 at 1 23 51 AM

**Pull request closed without merging**